buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -696,6 +696,7 @@ class BuildPlugin implements Plugin<Project> {
                 nonInputProperties.systemProperty('java.io.tmpdir', test.workingDir.toPath().resolve('temp'))
 
                 nonInputProperties.systemProperty('compiler.java', "${-> BuildParams.compilerJavaVersion.majorVersion}")
+                nonInputProperties.systemProperty('runtime.java', "${-> BuildParams.runtimeJavaVersion.majorVersion}")
```

Comment:
FWIW, we can now remove the `->` bit here. That indicates a "lazy" gstring, which is no longer necessary since both runtime and compiler java versions are now accessible at configuraiton time.

Reply:
If these are the only two uses to remove, happy to do this in this PR